src/browser.ts

```diff
@@ -37,102 +33,94 @@ export const ALL_SCOPES: string[] = [];
-      for (const methodName of methods) {
-          defaults[methodName] = settings;
-      }
-
-      return defaults;
-  }
-
-  async createStub(
-      service: protobuf.Service,
-      opts: ClientStubOptions) {
-          const authHeader = await this.auth.getRequestHeaders();
-          async function serviceClientImpl(method, requestData, callback) {
-
-               let headers = Object.assign({}, authHeader);
-               headers['Content-Type'] = 'application/x-protobuf';
-               headers['User-Agent'] = 'testapp/1.0';
-
-               // method.parent.options['(google.api.default_host)'] gives us the api.googleapis.com part of the url
-               // method.parent.parent.options.java_package.substring(4) gives is google.cloud.api.v1 part of the url
-               // method.parent.name gives us the service and method.name gives us the method being used
-               const url =
-               'https://' + opts['servicePath'] + '/$rpc/' +
-               method.parent.parent.options.java_package.substring(4) + '.' + method.parent.name + '/' + method.name;
-   
-               const fetchResult = await fetch(url, {
-                 headers,
-                 method: 'post',
-                 body: requestData,
-               });
-         
-               if (!fetchResult.ok) {
-                 callback(new Error(fetchResult.statusText));
-                 return;
-               }
-               const responseArrayBuffer = await fetchResult.arrayBuffer();           
-               callback(null, new Uint8Array(responseArrayBuffer));
-         }
-
-         const languageServiceStub = service.create(serviceClientImpl, false, false);
-
-         const methods = this.getServiceMethods(service);
-
-         const newLanguageServiceStub = {};
-         for (const methodName of methods) {
-           newLanguageServiceStub[methodName] = (req, options, metadata, callback) => {
-             languageServiceStub[methodName].apply(languageServiceStub, [req, callback]);
-           };
-         }
-
-         return newLanguageServiceStub;
-  }
+    auth: GoogleAuth;
+    promise?: PromiseConstructor;
+
+    constructor(opts, settings?: any) {
+        settings = settings || {};
+        this.auth = opts.auth;
+        this.promise = 'promise' in settings ? settings.promise! : Promise;
+    }
+
+    loadProto(jsonObject) {
+        const rootObject = protobuf.Root.fromJSON(jsonObject);
+        return rootObject;
+    }
+
+    private getServiceMethods(
+        service: protobuf.Service
+    ) {
+        const methods = Object.keys(service.methods);
+
+        // Creating an array of all methods within the stub with correct case (i.e, first letter is lower-case)
+        const methodsCorrectCase : string[] = [];
+        for (const methodName of methods) {
+            methodsCorrectCase.push(methodName.substring(0,1).toLowerCase() + methodName.substring(1));
```

Comment:
Last comment, I promise! Sorry for missing this yesterday.

Let's make this more readable by doing

```
const methodCorrectCase = methodName[0].toLowerCase() + methodName.substring(1);
methodsCorrectCase.push(methodCorrectCase);
```

Alternatively, we can go functional and use `map`:

```
const methodsCorrectCase = methods.map(method => {
  return method[0].toLowerCase() + method.substring(1);
});
```

Whichever option you prefer, it's better than having a non-trivial expression in a function parameter.